psi4/driver/qcdb/vecutil.py

```diff
@@ -75,12 +76,12 @@ def naivemult(v, u):
 def normalize(v):
     """Compute normalized vector *v*."""
     vmag = norm(v)
-    return [v[i] / vmag  for i in range(len(v))]
```

Comment:
Any reason not to use NumPy here? Especially for distance/cross.

Reply:
I thought a NumPy renovation, too, of this somewhat shameful file. But the overwhelming preponderance of "vectors" that hit this are length 3. I doubt its the best that can be done, but I did a medium-scale optimization of Molecule mid-Dec and ended up adding and then removing numpy in various places. I think the `distance` expression is probably "optimized" for its role. `cross` is used less often. I can switch it out with the numpy version if you think it'd look less naive.

Reply:
If these are pure vector3 arrays, NumPy is hard due to its inherent overhead. Really the only low latency (I will define this as "time to C" here) function in NumPy is `einsum` which I made less low-latency lately. Could we simply state that we are using pure Python here as we *expect* these arrays to be extremely small?